README.md

```diff
@@ -2665,24 +2665,37 @@ Other Style Guides
   <a name="semicolons--required"></a><a name="20.1"></a>
   - [21.1](#semicolons--required) **Yup.** eslint: [`semi`](http://eslint.org/docs/rules/semi.html) jscs: [`requireSemicolons`](http://jscs.info/rule/requireSemicolons)
 
+    > Why? When JavaScript encounters a line break without a semicolon, it uses a set of rules called [Automatic Semicolon Insertion](https://www.ecma-international.org/ecma-262/5.1/#sec-7.9) to determine whether or not it should regard that line break as the end of a statement, and (as the name implies) place a semicolon into your code before the line break if it thinks so. ASI contains a few eccentric behaviors, though, and your code will break if JavaScript misinterprets your line break. These rules will become more complicated as new features become a part of JavaScript. Explicitly terminating your statements and configuring your linter to catch missing semicolons will help prevent you from encountering issues.
+
     ```javascript
-    // bad
-    (function () {
-      const name = 'Skywalker'
-      return name
-    })()
+    // bad - raises exception
+    const luke = {}
+    const leia = {}
+    [luke, leia].forEach(jedi => jedi.father = 'vader')
+
+    // bad - raises exception
+    const foo = 2
+    ("No! That's impossible!".repeat(foo))
+
+    // bad - returns `undefined` instead of the value on the next line - **always** happens when `return` is on a line by itself because of ASI!!
+    function foo() {
+      return
+        "search your feelings, you know it to be foo"
+    }
 
     // good
-    (function () {
-      const name = 'Skywalker';
-      return name;
-    }());
+    const luke = {};
+    const leia = {};
+    [luke, leia].forEach(jedi => jedi.father = 'vader');
```

Comment:
this is a bad example; elsewhere in the guide we require that this arrow function be surrounded by curly braces

Reply:
Ah, right, because mutations are side effects 😅

Reply:
assignments aren't mutations tho :-)

Reply:
Thank you. Semantics are difficult but quite important ☝️

Reply:
Although, it seems somewhat odd to consider an alteration to an object that will alter the return of both `obj.hasOwnProperty(key)` and `Object.keys(obj)` as **not** a mutation to the structure of the object. A very fast Google of the subject implies that [doing exactly this is, indeed, a mutation](https://slemgrim.com/mutate-or-not-to-mutate/). Perhaps I need more research.

Reply:
Further research of [Mozilla's documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze) of how `Object.freeze` makes objects _im_-mutable implies that creating a new property on one, or changing a property already within one, is indeed a mutation

Reply:
@thepeoplesbourgeois ha, touché. `jedi.father =` is indeed a mutation :-)

however the reason this style isn't permitted isn't because it's a mutation - `jedi => Object.assign(jedi, { father: 'vader' })` would be fine - but because it's an assignment statement.